[CI] Change tests used by the new dashboard

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96986

Summary: Stop using runn.py to trigger the new dashboard run. Instead,
we spell out the actual cmd which will be easier to extend. Dropping
perf tests for dynamo_eager and aot_eager in this PR.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

desertfire committed:
[CI] Change tests used by the new dashboard

Summary: Stop using runn.py to trigger the new dashboard run. Instead,
we spell out the actual cmd which will be easier to extend. Dropping
perf tests for dynamo_eager and aot_eager in this PR.

[ghstack-poisoned]

desertfire committed:
Update on "[CI] Change tests used by the new dashboard"


Summary: Stop using runn.py to trigger the new dashboard run. Instead,
we spell out the actual cmd which will be easier to extend. Dropping
perf tests for dynamo_eager and aot_eager in this PR.

[ghstack-poisoned]

huydhn commented on .ci/pytorch/test.sh:
```diff
@@ -848,7 +884,7 @@ elif [[ "${TEST_CONFIG}" == *dynamo* && "${SHARD_NUMBER}" == 2 && $NUM_TEST_SHAR
   install_torchvision
   install_huggingface
-  test_dynamo_benchmark huggingface ""
+  test_dynamo_benchmark huggingface "$id"
```
You would need to add `id=$((SHARD_NUMBER-1))` here and below, otherwise `$id` would be undefined

huydhn commented on .ci/pytorch/test.sh:
```diff
@@ -278,6 +278,46 @@ else
+    # Run accuracy test
+    # All the accuracy tests can be skipped once the CI accuracy checking is stable enough
+    for backend in eager aot_eager; do
+      python "benchmarks/dynamo/$suite.py" \
```
Just curious, where are the performance tests for `eager` and `aot_eager`?  I see only performance tests for `inductor` below.  They are not needed?

huydhn commented on .ci/pytorch/test.sh:
```diff
@@ -278,6 +278,46 @@ else
+        --output "$TEST_REPORTS_DIR/{$backend}_no_cudagraphs_{$suite}_{$dtype}_training_cuda_accuracy.csv"
+    python "benchmarks/dynamo/$suite.py" \
+        --accuracy --backend "$backend" "$@" \
+        --output "$TEST_REPORTS_DIR/{$backend}_with_cudagraphs_{$suite}_{$dtype}_training_cuda_accuracy.csv"
```
Note: I will update the Rockset query accordingly, so now we have `inductor_no_cudagraphs` and `inductor_with_cudagraphs`

desertfire committed:
Update on "[CI] Change tests used by the new dashboard"


Summary: Stop using runn.py to trigger the new dashboard run. Instead,
we spell out the actual cmd which will be easier to extend. Dropping
perf tests for dynamo_eager and aot_eager in this PR.

[ghstack-poisoned]

huydhn reviewed (approved):
LGTM!

desertfire committed:
Update on "[CI] Change tests used by the new dashboard"


Summary: Stop using runn.py to trigger the new dashboard run. Instead,
we spell out the actual cmd which will be easier to extend. Dropping
perf tests for dynamo_eager and aot_eager in this PR.

[ghstack-poisoned]

weiwangmeta reviewed (approved):
Looks good to me! 
My understanding is we may need to further increase the number of runners if this gets merged. 12 may not serve two 8 shard jobs (one nightly + one experiment PR).

desertfire committed:
Update on "[CI] Change tests used by the new dashboard"


Summary: Stop using runn.py to trigger the new dashboard run. Instead,
we spell out the actual cmd which will be easier to extend. Dropping
perf tests for dynamo_eager and aot_eager in this PR.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx

[ghstack-poisoned]

weiwangmeta commented on .github/workflows/inductor-perf-test-nightly.yml:
```diff
@@ -2,7 +2,7 @@ name: inductor-A100-perf-nightly
 on:
   schedule:
-    - cron: 45 1 * * *
+    - cron: 45 1,7,13,19 * * *
```
Is 4 times a day too frequent? Maybe we can set to 2 and see if it gives sufficient amount of information that is needed. 
If the job is producing more information than we can consume, or if there are less amount of delta, e.g. two adjacent night runs where the commits may be identical or closely identical, there may not be a need to run 4 times. 2 or 3 may be a good balance?

weiwangmeta replied:
Sounds good.

desertfire commented:
@pytorchbot merge -f "Previously inductor shards have all passed. inductor_timm_dynamic test failure is flaky."
